docs/standard/serialization/snippets/system-text-json-how-to/csharp/RoundtripDataTable.cs

```diff
@@ -0,0 +1,246 @@
+                    }
+                    firstPass = false;
+                }
+                var row = dataTable.NewRow();
+                foreach (var col in element.EnumerateObject())
+                {
+                    row[col.Name] = col.Value.JsonElementToTypedValue();
+                }
+                dataTable.Rows.Add(row);
+            }
+            return dataTable;
+        }
+
+        private static Type ValueKindToType(this JsonValueKind valueKind, string value)
+        {
+            switch (valueKind)
+            {
+                case JsonValueKind.String:      // 3
+                    return typeof(System.String);
+                case JsonValueKind.Number:      // 4    
+                    if (Int64.TryParse(value, out var intValue))
+                    {
+                        return typeof(System.Int64);
+                    }
+                    else
+                    {
+                        return typeof(System.Double);
+                    }
+                case JsonValueKind.True:        // 5
+                case JsonValueKind.False:       // 6
+                    return typeof(System.Boolean);
+                case JsonValueKind.Undefined:   // 0
+                    return null;
+                case JsonValueKind.Object:      // 1 
+                    return typeof(System.Object);
+                case JsonValueKind.Array:       // 2
+                    return typeof(System.Array);
+                case JsonValueKind.Null:        // 7
+                    return null;
+                default:
+                    return typeof(System.Object);
+            }
+        }
+
+        private static object JsonElementToTypedValue(this JsonElement jsonElement)
+        {
+            switch (jsonElement.ValueKind)
+            {
+                case JsonValueKind.Object:      // 1  (these need special handling)?
+                case JsonValueKind.Array:       // 2
+                case JsonValueKind.String:      // 3
+                    if (jsonElement.TryGetGuid(out Guid guidValue))
+                    {
+                        return guidValue;
+                    }
+                    else
+                    {
+                        if (jsonElement.TryGetDateTime(out DateTime datetime))
+                        {
+                            // If an offset was provided, use DateTimeOffset.
+                            if (datetime.Kind == DateTimeKind.Local)
+                            {
+                                if (jsonElement.TryGetDateTimeOffset(out DateTimeOffset datetimeOffset))
+                                {
+                                    return datetimeOffset;
+                                }
+                            }
+                            return datetime;
+                        }
+                        return jsonElement.ToString();
```

Comment:
What's the behavior for arrays and object? I think it's better to throw `NotSupportedException` and let users decide when they fork our example (I'd also do this anywhere it's not clear what the expected behavior should be).